Sources/Nimble/Matchers/MatchError.swift

```diff
@@ -6,16 +6,24 @@ import Foundation
 /// Errors are tried to be compared by their implementation of Equatable,
 /// otherwise they fallback to comparison by _domain and _code.
 public func matchError<T: Error>(_ error: T) -> Predicate<Error> {
-    return Predicate.fromDeprecatedClosure { actualExpression, failureMessage in
-        let actualError: Error? = try actualExpression.evaluate()
+    return Predicate.define { actualExpression in
+        let actualError = try actualExpression.evaluate()
+
+        let failureMessage = FailureMessage()
+        setFailureMessageForError(
```

Comment:
Would it make sense for this function to become a mutating method on `FailureMessage`?

Reply:
`FailureMessage` is a class that extends `NSObject` so that does not make sense. Instead, I would like to replace the method with one which returns `ExpectationMessage`.